src/mainviews/job-detail.tsx

```diff
@@ -1,13 +1,398 @@
+import Box from '@mui/material/Box';
+import Breadcrumbs from '@mui/material/Breadcrumbs';
+import Link from '@mui/material/Link';
+import Stack from '@mui/material/Stack';
+import Button from '@mui/material/Button';
+import Accordion from '@mui/material/Accordion';
+import {
+  AccordionDetails,
+  AccordionSummary,
+  Checkbox,
+  CircularProgress,
+  FormControl,
+  FormControlLabel,
+  FormGroup,
+  FormLabel,
+  OutlinedInputProps,
+  TextField,
+  Typography
+} from '@mui/material';
+import { caretDownIcon } from '@jupyterlab/ui-components';
+import { useTranslator } from '../hooks';
+import { Heading } from '../components/heading';
+import { Scheduler, SchedulerService } from '../handler';
+import { JupyterFrontEnd } from '@jupyterlab/application';
 
 export interface IJobDetailProps {
+  app: JupyterFrontEnd;
   model: IJobDetailModel;
   modelChanged: (model: IJobDetailModel) => void;
+  setCreateJobModel: (createModel: ICreateJobModel) => void;
   setView: (view: JobsView) => void;
 }
 
+interface ITextFieldStyledProps {
+  label: string;
+  defaultValue?: string;
+  InputProps?: Partial<OutlinedInputProps> | undefined;
+}
+
 export function JobDetail(props: IJobDetailProps): JSX.Element {
-  return <h1>Hi there!</h1>;
+  const [loading, setLoading] = useState(true);
+  const [job, setJob] = useState<Scheduler.IDescribeJob | undefined>(undefined);
+
+  //TO DELETE
+  const prop = { model: { jobId: '3f062962-1e3e-442b-8454-e76af250da39' } };
+
+  const trans = useTranslator('jupyterlab');
+
+  const ss = new SchedulerService({});
+
+  const getJobDefinion = async () => {
+    const jobDefinition = await ss.getJob(prop.model.jobId);
+    setJob(jobDefinition);
+    setLoading(false);
+  };
+
+  // Retrieve the key from the parameters list or return a parameter with a null value
+  const getParam = (key: string) => {
+    return {
+      name: key,
+      value: job?.parameters?.[key]
+    };
+  };
+
+  const rerunJob = () => {
+    const initialState: ICreateJobModel = {
+      inputFile: job?.input_uri ?? '',
+      jobName: job?.name ?? '',
+      outputPath: '.',
+      environment: job?.runtime_environment_name ?? '',
+      parameters:
+        job?.parameters !== undefined
+          ? Object.keys(job.parameters).map(key => getParam(key))
+          : undefined,
+      outputFormats: job?.output_formats?.map(format => ({
+        name: format,
+        label: format
+      }))
+    };
+
+    props.setCreateJobModel(initialState);
+    props.setView('CreateJob');
+  };
+
+  const handleDeleteJob = async () => {
+    await ss.deleteJob(job?.job_id ?? '');
+    props.setView('ListJobs');
+  };
+
+  const handleStopJob = async () => {
+    props.app.commands.execute('scheduling:stop-job', {
+      id: job?.job_id
+    });
+    getJobDefinion();
+  };
+
+  useEffect(() => {
+    getJobDefinion();
+  }, []);
+
+  function TextFieldStyled(props: ITextFieldStyledProps) {
+    return (
+      <TextField
+        {...props}
```

Comment:
Would this be a read-only field? Should read-only fields be styled any differently than editable fields?

Reply:
For this release they are read-only, for future it is planned that we will allow users to edit all fields at this view (as far as I understand). We can make them `disabled` instead of `read-only` as they are now. When disabled, elements are greyed-out and are not registering interaction events:
![Screen Shot 2022-09-26 at 1 44 43 PM](https://user-images.githubusercontent.com/26686070/192377093-4e423df2-0811-4429-a8df-d01291a358d5.png)

After additional consideration, `disabled` is probably a better option than `read-only` for UI/UX clarity for this release.